doc/user/content/sql/create-source/webhook.md

```diff
@@ -157,3 +157,128 @@ After a successful secret creation, you can use the same secret to create differ
 ```
 
 Your new webhook is now up and ready to accept requests using the basic authentication.
+
+### Connecting with Segment
+
+[Segment](https://segment.com/) is a commonly used tool for collecting events from your
+applications. You can supercharge these events by ingesting them into Materialize and joining it
+with your other data!
+
+The first step for setting up a webhook source is to create a shared secret. While this isn't
+required, it's the recommended best practice.
+
+```
+CREATE SECRET segment_shared_secret AS 'abc123';
+```
+
+Using this shared key, Segment will sign each request and we can use the signature to determine if
+the request is legitmate.
+
+After defining a shared secret, we can create the source itself:
+
+```
+CREATE SOURCE my_segment_source IN CLUSTER my_cluster FROM WEBHOOK
+  BODY FORMAT JSON
+  INCLUDE HEADERS
+  CHECK (
+    WITH ( BODY BYTES, HEADERs, SECRET segment_shared_secret AS secret BYTES)
+    decode(headers->'x-signature', 'hex') = hmac(body, secret, 'sha1')
+  )
+```
+
+This creates a source called _my_segment_source_ and installs it in cluster named _my_cluster_.
+The source will have two columns, _body_ of type `jsonb` and _headers_ of type `map[text=>text]`.
+
+The `CHECK` statement defines how to validate each request. At the time of writing, Segment
+validates requests by signing them with an HMAC in the `X-Signature` request header. The HMAC is a
+hex-encoded SHA1 hash using the shared secret and request body. We can decode the signature using
+the [`decode`](/sql/functions/#decode) function, getting the raw bytes, and generate our own HMAC
+using the [`hmac`](/sql/functions/#hmac) function. If the two values are equal, then the request is
+legitimate!
+
+{{< note >}}
+
+For the latest information on Segment's Webhook Destination, please see their
+[documentation](https://segment.com/docs/connections/destinations/catalog/actions-webhook/).
+
+{{< /note >}}
+
+### Duplicated and Partial Events
```

Comment:
Is this a pattern that developers should always follow as a good practice? I believe it (and also `## Debugging CHECK Statements`) deserves its own section before talking about them in the `## Examples` section, which should include a reference to a smaller example within the `## Examples` section

E.g.:

```md
...
### Debugging `CHECK` statements
At first, getting your `CHECK` statement correct can be difficult...

### Duplicated and Partial Events
Given any number of conditions ...
...

## Examples
### Creating a Basic Authentication
...
### Creating a Source
...
### Debugging `CHECK` statements
...
### Deduplicating events
...
```

Reply:
Good callout! I moved both sections out of "Examples"